File: Serveur/src/main/java/com/arepo/serveur/model/ConventionCadre.java
package com.arepo.serveur.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDate;
import java.util.UUID;


@Entity
@Table(name = "convention_cadre")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ConventionCadre {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id_convention_cadre")
    private UUID idConventionCadre;

    @Column(name = "montant_contribution")
    private Double montantContribution;

    @Column(name = "montant_debloque")
    private Double montantDebloque;

    @Column(name = "etat_convention", nullable = false, length = 30)
    private String etatConvention;

    @Column(name = "date_participation")
    private LocalDate dateParticipation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_partenaire", referencedColumnName = "id_partenaire", nullable = false)
    private Partenaire partenaire;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_programme", referencedColumnName = "id_programme", nullable = false)
    private Programme programme;
}

File: Client/src/types/conventionCadre.ts
export interface ConventionCadre {
  idConventionCadre: string
  montantContribution: number | null
  montantDebloque: number | null
  etatConvention: string
  dateParticipation: string | null
  idPartenaire: string   // ← juste l'UUID, plus d'objet imbriqué
  idProgramme: string
}

export interface ConventionCadreCreateRequest {
  idPartenaire: string
  idProgramme: string
  montantContribution: number
  montantDebloque: number
  etatConvention: string
  dateParticipation: string | null
}